buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -131,6 +133,7 @@ class ClusterFormationTasks {
     /** Adds a task to extract the elasticsearch distribution */
     static Task configureExtractTask(String name, Project project, Task setup, NodeInfo node) {
         List extractDependsOn = [project.configurations.elasticsearchDistro, setup]
+        // TODO assert that project.configurations.elasticsearchDistro.singleFile is from a project and not the cache
```

Comment:
But you are suggesting making this required (being a substitution). This is an edge case because we hacked the integration test cluster to run with something other than a simple zip.  I don't think we should break external plugins (requiring them to run with an elasticsearch checkout to do *anything*). I'm sorry you spent time on this.